src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -420,6 +547,49 @@ private ODataNestedResourceInfo ReadNestedResourceInfo()
             return link;
         }
 
+        /// <summary>
+        /// Reads an <see cref="ODataDeletedResource"/>
+        /// </summary>
+        /// <returns>An <see cref="MaterializerDeletedEntry"/> of the read <see cref="ODataDeletedResource"/>.</returns>
+        private MaterializerDeletedEntry ReadDeletedResource()
+        {
+            this.ExpectState(ODataReaderState.DeletedResourceStart);
+
+            ODataDeletedResource result = (ODataDeletedResource)this.reader.Item;
+            MaterializerDeletedEntry entry;
+
+            Debug.Assert(this.items != null, "this.items should not be null");
+
+            entry = MaterializerDeletedEntry.CreateDeletedEntry(
+                    result,
+                    this.materializerContext);
+
+            if (this.items.Count > 0)
+            {
+                this.items.Push(entry);
+            }
+
+            do
+            {
+                this.AssertRead();
+
+                switch (this.reader.State)
+                {
+                    case ODataReaderState.DeletedResourceEnd:
+                        if (this.items.Count > 0)

```

Comment:
Given my argument [here](https://github.com/OData/odata.net/pull/2627/files#r1172097526), should we find ourselves here if there isn't an item in the stack?

Reply:
fixed